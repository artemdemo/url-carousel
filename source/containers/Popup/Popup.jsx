import React from 'react';

import './Popup.less';

const Popup = () => {
    return (
        <div className='browser-menu'>
            <ul className='list-group'>
                <li className='list-group-item'>Start carousel</li>
                <li className='list-group-item'
                    onClick={() => chrome.runtime.openOptionsPage()}>
                    Settings
                </li>
            </ul>
        </div>
    );
};

export default Popup;
