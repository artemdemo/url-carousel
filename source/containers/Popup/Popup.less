@bg-hover: #ebebeb;

.browser-menu {
    width: 250px;
}

.browser-menu-list {
    list-style: none;
    padding: 0;
    margin: 0;
}
    .browser-menu-list__item {
        padding: 8px 10px;
        cursor: pointer;
        border-bottom: 1px solid darken(@bg-hover, 5%);

        &:hover {
            background-color: @bg-hover;
        }

        &:last-child {
            border-bottom: none;
        }
    }
