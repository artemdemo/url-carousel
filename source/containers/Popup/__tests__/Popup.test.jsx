import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Popup from '../Popup';

describe('<Popup>', () => {
    it('Render', () => {
        const tree = renderer.create(
            <MuiThemeProvider>
                <Popup />
            </MuiThemeProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
