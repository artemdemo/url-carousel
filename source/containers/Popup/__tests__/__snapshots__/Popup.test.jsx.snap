// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup> Render 1`] = `
<div
  className="browser-menu"
>
  <div
    style={
      Object {
        "muiPrepared": true,
        "padding": "8px 0px 8px 0px",
      }
    }
  >
    <div>
      <span
        disabled={undefined}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 72,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fill": "#757575",
                  "height": 24,
                  "left": 4,
                  "margin": 12,
                  "muiPrepared": true,
                  "position": "absolute",
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M8 5v14l11-7z"
              />
            </svg>
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              Start carousel
            </div>
          </div>
        </div>
      </span>
    </div>
    <div>
      <span
        disabled={undefined}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 72,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fill": "#757575",
                  "height": 24,
                  "left": 4,
                  "margin": 12,
                  "muiPrepared": true,
                  "position": "absolute",
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
              />
            </svg>
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              Settings
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
